x-pack/auditbeat/module/system/login/login.go

```diff
@@ -0,0 +1,193 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build linux,cgo
+
+package login
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/auditbeat/datastore"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+const (
+	moduleName    = "system"
+	metricsetName = "login"
+
+	bucketName = "auditbeat.login.v1"
+
+	eventTypeEvent = "event"
+
+	// LoginRecordTypeBoot represents a system boot.
+	LoginRecordTypeBoot = "boot"
+	// LoginRecordTypeShutdown represents a system shutdown (halt or reboot).
+	LoginRecordTypeShutdown = "shutdown"
+	// LoginRecordTypeUserLogin represents a user login.
+	LoginRecordTypeUserLogin = "user_login"
+	// LoginRecordTypeUserLogout represents a user logout.
+	LoginRecordTypeUserLogout = "user_logout"
+)
+
+// LoginRecord represents a login record.
+type LoginRecord struct {
+	Utmp      Utmp
+	Type      string
+	PID       int
+	TTY       string
+	UID       int
+	Username  string
+	Hostname  string
+	IP        *net.IP
+	Timestamp time.Time
+	Origin    string
+}
```

Comment:
Same, long name and a many exported symbols that we might not need somewhere else.

Reply:
`LoginRecord` is encoded and decoded by `gob` so it needs to be exported, unfortunately.